dspace-api/src/main/java/org/dspace/handle/HandleClarinServiceImpl.java

```diff
@@ -0,0 +1,277 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.handle;
+
+import java.sql.SQLException;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.Logger;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.content.MetadataFieldServiceImpl;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.core.LogHelper;
+import org.dspace.handle.dao.HandleClarinDAO;
+import org.dspace.handle.dao.HandleDAO;
+import org.dspace.handle.service.HandleClarinService;
+import org.dspace.handle.service.HandleService;
+import org.dspace.services.ConfigurationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.ObjectUtils;
+
+/**
+ * Additional service implementation for the Handle object in Clarin-DSpace.
```

Comment:
Please, add more explanotory description. What does it do?

Reply:
Added explanation comment to the HandleClarinService class:
`Added External Handles, ability to update prefix for every Handle, generating the handle prefix based on the
pid configuration.`